File: src/web-client/src/api/requests/PostSignupRequest.ts
import BaseRequest, {RequestMethod} from "./BaseRequest";
import PostSignupResponse from "../responses/PostSignupResponse";

export type PostSignupRequestData = {
    username: string
    firstName: string
    lastName: string
    dateOfBirth: string
    gender: string
    email?: string | undefined
    password: string
}

class PostSignupRequest extends BaseRequest<PostSignupResponse, PostSignupRequestData> {

    constructor(data: PostSignupRequestData) {
        super(RequestMethod.POST, '/api/signup', data, {
            headers: {
                "Content-Type": "application/x-www-form-urlencoded"
            }
        });
    }

}

export default PostSignupRequest

File: src/main/java/dev/bayun/id/api/schema/request/PostSignupRequest.java
package dev.bayun.id.api.schema.request;

import dev.bayun.id.core.validation.annotation.*;
import lombok.Data;

@Data
public class PostSignupRequest {

    @Username
    private String username;

    @FirstName
    private String firstName;

    @LastName
    private String lastName;

    @DateOfBirth
    private String dateOfBirth;

    @Gender
    private String gender;

    @Password
    private String password;

    @Email(required = false)
    private String email;

}
